DPGAnalysis/Skims/src/JetHTJetPlusHOFilter.cc

```diff
@@ -0,0 +1,258 @@
+      jetdirection.push_back(etaphi);
+      jetspt.push_back((*PFJets)[jet].pt());
+      passed = true;
+    }
+  }
+  
+  if (!passed) return false;
+  Njetp++;
+  bool isJetDir=false;
+
+  edm::Handle<PFClusterCollection> hoht;
+  iEvent.getByToken(tok_hoht_,hoht);
+  if (hoht.isValid()) {
+    if ((*hoht).size()>0) {
+      for (PFClusterCollection::const_iterator ij=(*hoht).begin(); ij!=(*hoht).end(); ij++){
+	double hoenr = (*ij).energy();
+	if (hoenr <hothres) continue;
+   
+    const math::XYZPoint&  cluster_pos = ij->position();
+ 
+	
+	double hoeta = cluster_pos.eta() ;
+	double hophi = cluster_pos.phi() ;
+
+	for (unsigned ijet = 0; ijet< jetdirection.size(); ijet++) {
+	  double delta = delta2R(jetdirection[ijet].first, jetdirection[ijet].second, hoeta, hophi);
+	  if (delta <0.5) { 
+	    isJetDir=true;   break;
+	    
+	    
+	  }
+	}
+      }
+    }
+  }
+
+  if (isJetDir) {Npass++;}
+      
+  return isJetDir;
+
+}
+
+// ------------ method called once each job just before starting event loop  ------------
+void 
+JetHTJetPlusHOFilter::beginJob()
+{
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void 
+JetHTJetPlusHOFilter::endJob() {
+  cout<<"End of JetHTJetPlusHOFilter with event "<<Nevt<<" Jetpassed "<< Njetp<<" passed "<<Npass<<endl;
+
+
+}
+
+// ------------ method called when starting to processes a run  ------------
+bool 
+JetHTJetPlusHOFilter::beginRun(edm::Run&, edm::EventSetup const&)
+{ 
+  return true;
+}
+
+// ------------ method called when ending the processing of a run  ------------
+bool 
+JetHTJetPlusHOFilter::endRun(edm::Run&, edm::EventSetup const&)
+{
+  return true;
+}
+
+// ------------ method called when starting to processes a luminosity block  ------------
+bool 
+JetHTJetPlusHOFilter::beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+  return true;
+}
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+bool 
+JetHTJetPlusHOFilter::endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+  return true;
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+JetHTJetPlusHOFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+
+double JetHTJetPlusHOFilter::PhiInRange(const double& phi) {
+  double phiout = phi;
+  
+  if( phiout > 2*M_PI || phiout < -2*M_PI) {
+    phiout = fmod( phiout, 2*M_PI);
+  }
+  if (phiout <= -M_PI) phiout += 2*M_PI;
+  else if (phiout >  M_PI) phiout -= 2*M_PI;
+  
+  return phiout;
+}
+
+double JetHTJetPlusHOFilter::delta2R(double eta1, double phi1, double eta2, double phi2) {
```

Comment:
@kfjack - please use the already existing implementation in DataFormats/Math/interface/deltaR.